python/ray/serve/tests/test_proxy_state.py

```diff
@@ -41,84 +34,96 @@ def get_alive_node_ids(self):
-        (f"worker-node-id-{i}", f"fake-worker-ip-{i}")
-        for i in range(number_of_worker_nodes)
-    ]
+class FakeProxyWrapper(ProxyWrapper):
+    def __init__(self, *args, **kwargs):
+        self.actor_handle = FakeProxyActor(*args, **kwargs)
+        self.ready = ProxyWrapperCallStatus.FINISHED_SUCCEED
+        self.health = ProxyWrapperCallStatus.FINISHED_SUCCEED
+        self.worker_id = "mock_worker_id"
+        self.log_file_path = "mock_log_file_path"
+        self.health_check_ongoing = False
+        self.is_draining = False
+        self.shutdown = False
+        self.num_health_checks = 0
+
+    @property
+    def actor_id(self) -> str:
+        pass
 
+    def reset_health_check(self):
+        pass
 
-@pytest.fixture()
-def setup_controller():
-    try:
-        controller = ray.get_actor(SERVE_CONTROLLER_NAME, namespace=SERVE_NAMESPACE)
-    except ValueError:
-        controller = ServeController.options(
-            name=SERVE_CONTROLLER_NAME, namespace=SERVE_NAMESPACE
-        ).remote(
-            SERVE_CONTROLLER_NAME,
-            http_config=None,
-        )
-    controller_actor_id = controller._ray_actor_id.hex()
+    def start_new_ready_check(self):
+        pass
 
-    def check_controller_alive():
-        controller_actor_info = ray._private.state.actors(controller_actor_id)
-        controller_actor_state = controller_actor_info["State"]
-        return controller_actor_state == "ALIVE"
+    def start_new_health_check(self):
+        self.health_check_ongoing = True
 
-    wait_for_condition(check_controller_alive)
-    yield
+    def start_new_drained_check(self):
+        pass
 
+    def is_ready(self) -> ProxyWrapperCallStatus:
+        return self.ready
 
-@ray.remote(num_cpus=0)
-class MockProxyActor:
-    def __init__(self, *args, **kwargs):
+    def is_healthy(self) -> ProxyWrapperCallStatus:
+        self.num_health_checks += 1
+        self.health_check_ongoing = False
+        return self.health
+
+    def is_drained(self) -> ProxyWrapperCallStatus:
         pass
 
-    async def ready(self):
-        return json.dumps(["mock_worker_id", "mock_log_file_path"])
+    def is_shutdown(self):
+        return self.shutdown
 
-    async def check_health(self):
+    def update_draining(self, draining: bool):
         pass
 
+    def kill(self):
+        self.shutdown = True
+
+    def get_num_health_checks(self):
+        return self.num_health_checks
+
+
+def _create_proxy_state_manager(
+    http_options: HTTPOptions = HTTPOptions(),
+    head_node_id: str = HEAD_NODE_ID,
+    cluster_node_info_cache=MockClusterNodeInfoCache(),
+    actor_proxy_wrapper_class=FakeProxyWrapper,
+) -> (ProxyStateManager, ClusterNodeInfoCache):
+    return (
+        ProxyStateManager(
+            SERVE_CONTROLLER_NAME,
+            config=http_options,
+            head_node_id=head_node_id,
+            cluster_node_info_cache=cluster_node_info_cache,
+            proxy_actor_class=FakeProxyActor,
```

Comment:
looks like this can be totally removed now :)

Reply:
oh that's right, let me take it out, too 😄